### ISSUE TITLE
Other languages.

### ISSUE THREAD

**Issue by spwilson2** [UW-UPL/Sample-C-Program-gofish#3]
Do we want more example programs for different languages? If so, do we want then to tall be the same go-fish program so people can compare across languages? Or does having them be different programs, so you can show off more idiomatic uses of the language?

**Comment by LOZORD** [comment]
I don't know -- it really depends on what this repo/project is used for. My opinion is that this should be moved into either the Archive repo or the Talk repo's `archive` directory since this repo is no longer active.

**Comment by spwilson2** [comment]
I was thinking we might make separate repos for different languages and
possibly a master repo with them as submodules, although that could be
cruft. Github doesn't allow groups/repo sections within groups does it? :(

**Comment by LOZORD** [comment]
First, what is the motivation for doing this in different languages? This
code is only here for reference for the C talk Ricky and Nik gave.

Second, no, I don't think a "master repo" feature exists on GitHub, or git.
I think the best thing would be to have a single "gofish" repo with
directories for each different implementation.

If you want to start porting this code to other languages, I would
definitely help, but my opinion is that it would "feature creep"/cruft and
wouldn't serve any useful purpose.

**Comment by spwilson2** [comment]
Well in git you can do submodules, the problem is it would just be a normal git repo on github if there's no hierarchy for repos.

Also I'm good with just archiving this repo then. I assumed this was just code that we could point users to if they were trying to learn C.

**Comment by spwilson2** [comment]
Still would like to hear any feedback from anyone else, otherwise if you're good with archiving @LOZORD  I'll give it a few days so anyone can respond beforehand.

**Comment by LOZORD** [comment]
@Ricky54326 any input?

**Comment by jdetter** [comment]
Yeah you can archive, I basically rewrote the whole thing and it's still pretty much all shit. This was not a good example game for teaching C in my opinion because it's so object oriented.

If we still wanted to do code tutorials maybe we could have one big repo with examples? I think that would be cool. The base directory could be like:

```
UPL-Code-Tutorials/
          + Python/
          + Java/
          + C++/
          + ...
```

And all of the folders could be filled with simple programs in each language. Something like that

**Comment by jdetter** [comment]
Just having this one C example probably isn't that helpful anyway, if we had more examples with more languages that would be really cool IMO.

**Comment by LOZORD** [comment]
Tbh the interview prep repo could contain something along those lines.

**Comment by jdetter** [comment]
@LOZORD that's actually a really good idea too lol

**Comment by jdetter** [comment]
@spwilson2 @Ricky54326 what do you think?

**Comment by spwilson2** [comment]
Are you thinking the answers for interview prep questions could be in different languages (like they are now)? Or do you mean programs like this -- meant as a tutorial should be in interview prep?

**Comment by jdetter** [comment]
I'm thinking of making an repository of code examples for different languages. But we can also sort the interview prep stuff by language as well.

**Comment by spwilson2** [comment]
I'm not necessarily a fan of adding them to interview prep since I feel like if people are looking at code examples it could be for a lot of other reasons than interview prep. But definitely worth at least mentioning examples in the readme there.

**Comment by jdetter** [comment]
Lol no I'm saying completely separate from interview prep, if we had another repository called like `UPL-Code-Examples` where we could have small programs like this in different languages. I'm not saying add this to interview prep.

**Comment by jdetter** [comment]
I can see where you got confused lol

**Comment by spwilson2** [comment]
Down to /play clowntown

**Comment by spwilson2** [comment]
^Apparently not a sound for me.

**Comment by LOZORD** [comment]
I don't think there's a real need for code examples (especially without a certain focus like interview prep solutions), but if you guys do, go ahead. Like I said, I will definitely help out 😁

**Comment by spwilson2** [comment]
I think there definitely could be for C, bash, make, and maybe python. Since the first three you're just expected to know in the department, and resources can be difficult to find for a beginner.

Edit: Or just impossible to understand.

**Comment by LOZORD** [comment]
Ok, I do actually like that idea, kind of like a bootcamp for knowledge required for UW CS. If you guys want to come up with a list of languages/technologies you want to cover, I'll take a look and see what I can help with.

**Comment by spwilson2** [comment]
I think this will be worth talking about at the next coord meeting since I'm sure others will have more input and be able to shape this idea even more. It also would be worth considering if we could center a talk or series of talks around these topics, or something like learning to be comfortable in a *nix environment. Although that might come close to stepping on WACM's toes.

I'll add this as a topic for the next coord meeting.

**Comment by jdetter** [comment]
Even Java could be helpful considering people don't necessarily know how to properly use LinkedLists and Iterators and stuff like that.

**Comment by katamaritaco** [comment]
I think having a repo for code examples/resources for beginners could be nice. There are already a million resources for that kinda stuff, but 1) it would be great to have a 'UW - specific things to know list', and 2) it'd be fun to make tbh.

Like previously said it's assumed you know some things which just aren't really taught - especially bash, useful easy unix commands, make etc. Wacms linux tut is nice but only covers Linux.

Another thought: This would be a long shot, but if UPL could partner with some profs/cllasses for attending a UPL workshop (zB git/version control) and have them get some classroom incentive for it.

Get Outlook for Androidhttps://aka.ms/ghei36

On Tue, Aug 23, 2016 at 2:11 PM -0500, "John Detter" <notifications@github.com<mailto:notifications@github.com>> wrote:

Even Java could be helpful considering people don't necessarily know how to properly use LinkedLists and Iterators and stuff like that.

##

**Comment by LOZORD** [comment]
The UPL needs to gamify the learning experience 😎

**Comment by jdetter** [comment]
I agree, I think we should do easy games like TicTacToe or like connect four for starts because they are simple and new programmers will probably be able to understand them better.

I could also help if anyone needs any assets for a game!

**Comment by LOZORD** [comment]
But whoever implements these should be sure to **expose the importance of the languages**. E.g. the bash module should show how to run commands, use basic logic (e.g. `if` and `for`), and do file I/O. The C one should use `malloc`/`free` extensively, and maybe even `fork`. If we do a M A T L A B one, graphs and or image I/O should be involved, for example.

**Comment by LOZORD** [comment]
Actually, designing these modules as games might be a good idea because then we avoid accidentally duplicating projects (A C A D E M I C D I S H O N E S T Y). For example, 354 (and 537?) lays claim to "implement a linked list in C".

**Comment by jdetter** [comment]
@LOZORD That's a really good point. We will have to make sure to use implementations that aren't `copy and paste` solutions to potential class projects/homework.
